src/com/maddyhome/idea/vim/group/MotionGroup.java

```diff
@@ -1046,23 +1084,44 @@ public int moveCaretToMiddleColumn(@NotNull Editor editor) {
     return moveCaretToColumn(editor, Math.max(0, Math.min(len - 1, width)), false);
   }
 
-  public int moveCaretToColumn(@NotNull Editor editor, int count, boolean allowEnd) {
-    int line = editor.getCaretModel().getLogicalPosition().line;
+  public int moveCaretToColumn(@NotNull Editor editor, @NotNull Caret caret, int count, boolean allowEnd) {
+    int line = caret.getLogicalPosition().line;
     int pos = EditorHelper.normalizeColumn(editor, line, count, allowEnd);
 
     return editor.logicalPositionToOffset(new LogicalPosition(line, pos));
   }
 
-  public int moveCaretToLineStartSkipLeading(@NotNull Editor editor) {
-    int logicalLine = editor.getCaretModel().getLogicalPosition().line;
+  /**
+   * @deprecated To move the caret, use {@link #moveCaretToColumn(Editor, Caret, int, boolean)}
+   */
+  public int moveCaretToColumn(@NotNull Editor editor, int count, boolean allowEnd) {
```

Comment:
This method is used in movements to the first/middle/last screen column methods, which doesn't have a version with multiple carets in this PR because they are a bit buggy. Is it better to give the primary caret in their calls?

Reply:
Yes, I believe it is, just to make it clear that we have some places in our codebase that work only with the primary caret. It is worth adding a TODO item for all the places where it's a temporary workaround.